sdk/python/kfp/components/structures.py

```diff
@@ -674,25 +674,27 @@ def outputs_dict_from_components_dict(
                 for name, parameter_dict in all_outputs.items()
             }
 
-        def extract_description_from_command(commands: List[str]) -> str:
-            for command in commands:
-                if isinstance(command, str) and 'import kfp' in command:
-                    for node in ast.walk(ast.parse(command)):
-                        if isinstance(
-                                node,
-                            (ast.FunctionDef, ast.ClassDef, ast.Module)):
-                            docstring = ast.get_docstring(node)
-                            if docstring:
-                                return docstring
-            return ''
+        def extract_description_from_command(
+                commands: List[str]) -> Union[str, None]:
+            if commands:
+                for command in commands:
```

Comment:
nit: the above two lines can be combined into 
```
for command in (commands or []):
```
for shorter code and less nesting.

(If type annotation here is trustworthy, then commands would never be `None` and you could skip the `or []` part).

Reply:
Good suggestion, @chensun! Thanks for helping me clean this up. I've updated.

The type annotations are trustworthy (see line 551). I've opted to handle the `None` command case in line 551 rather than in the function body to simplify the function interface.